pkg/controllers/secretstore/client_manager.go

```diff
@@ -216,29 +217,32 @@ func (m *Manager) shouldProcessSecret(store esv1beta1.GenericStore, ns string) (
 		return true, nil
 	}
 
-	namespaceList := &v1.NamespaceList{}
+	namespace := v1.Namespace{}
+	if err := m.client.Get(context.Background(), client.ObjectKey{Name: ns}, &namespace); err != nil {
+		return false, fmt.Errorf("failed to get a namespace %q: %w", ns, err)
+	}
 
+	nsLabels := labels.Set(namespace.GetLabels())
 	for _, condition := range store.GetSpec().Conditions {
+		var labelSelectors []*metav1.LabelSelector
 		if condition.NamespaceSelector != nil {
-			namespaceSelector, err := metav1.LabelSelectorAsSelector(condition.NamespaceSelector)
-			if err != nil {
-				return false, err
-			}
-
-			if err := m.client.List(context.Background(), namespaceList, client.MatchingLabelsSelector{Selector: namespaceSelector}); err != nil {
```

Comment:
Listing namespaces for every namespace selector is wasteful since we only care about one namespace, `ns.` Instead, we should get the namespace first and compare its labels with the conditions.